### DISCUSSION TITLE
Scopes

### DISCUSSION THREAD

**Discussion by adrienpoly** [excid3/madmin#30] [Q&A]
Hello 

Is there a way to generate a dashboard for a scoped resource ?

Lets take an example a `User` must be manually confirmed. Can I simply make a link to this scoped resource in the navigation, scope it in the controller and put custom actions in the index view (confirm/reject).

Last time I tried with Administrate the only option was to create a new model UserPending with a default scope in it. But that wasn't really convenient.

Thanks

**Comment by excid3** [comment]
I think we may already have partially what you need. You can define `scope` in your Resource and it should show up as a filter on the index page.

```ruby
class UserResource < Madmin::Resource
  scope :unconfirmed
end
```

The `scope` method: https://github.com/excid3/madmin/blob/master/lib/madmin/resource.rb#L25

The index view renders scopes here: https://github.com/excid3/madmin/blob/master/app/views/madmin/application/index.html.erb#L11

One thing this doesn't support is scopes with arguments though. I think that's to be expected though.

And we do need to have support for custom pages and things more in the future.

**Reply by adrienpoly** [reply]
cool will try that
